src/mjolnir/util.cc

```diff
@@ -48,6 +49,80 @@ const std::string old_to_new_file = "old_nodes_to_new_nodes.bin";
 const std::string intersections_file = "intersections.bin";
 const std::string shapes_file = "shapes.bin";
 
+void set_checksums(std::vector<std::string> paths, const std::string& tile_dir) {
+  std::sort(paths.begin(), paths.end());
+
+  // uses openssl's API which can build the digest from byte chunks to save memory
+  EVP_MD_CTX* ctx = EVP_MD_CTX_new();
+  std::array<unsigned char, 16> digest{};
+  try {
+    EVP_DigestInit_ex(ctx, EVP_md5(), nullptr);
+
+    std::vector<char> buffer(1 << 26); // 64 MiB
+
+    for (const auto& p : paths) {
+      std::ifstream in(p, std::ios::binary);
+      if (!in) {
+        throw std::runtime_error("Failed to open: " + p);
+      }
+
+      while (in) {
+        in.read(buffer.data(), static_cast<std::streamsize>(buffer.size()));
+        std::streamsize got = in.gcount();
+        if (got > 0) {
+          if (EVP_DigestUpdate(ctx, buffer.data(), static_cast<size_t>(got)) != 1) {
+            throw std::runtime_error("EVP_DigestUpdate failed");
+          }
+        }
+      }
+    }
+
+    unsigned int out_len = 0;
+    if (EVP_DigestFinal_ex(ctx, digest.data(), &out_len) != 1 || out_len != digest.size()) {
+      throw std::runtime_error("EVP_DigestFinal_ex failed");
+    }
+  } catch (...) {
+    EVP_MD_CTX_free(ctx);
+    throw;
+  }
+
+  EVP_MD_CTX_free(ctx);
+
+  // roll the 128 bit digest into a uint64
+  uint64_t lo = 0, hi = 0;
+  for (int i = 0; i < 8; ++i) {
+    lo = (lo << 8) | digest[i];
+    hi = (hi << 8) | digest[8 + i];
+  }
+
+  // write the checksum to the tile headers
+  uint64_t checksum = lo ^ hi;
```

Comment:
a slightly more robust way to hash combine two values:

```suggestion
  uint64_t checksum = lo;
  checksum ^= hasher(v) + 0x9e3779b97f4a7c15ull + (checksum<<12) + (checksum>>4);
```

its a real shame to have to pick up a new dep, i cant believe boost didnt have something that could compute a simple checksum of some data. maybe they do but its still being voted on or something

Reply:
its for on the fly tiles fetching

Reply:
that's baldr no? or does some mjolnir executable do that which is not protected with `ENABLE_HTTP`?

Reply:
went with `boost::hash_combine`